# Week 1. Data Preprocessing: Data loading and cleaning

In [ ]:
#Importing necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [ ]:
# # Setting options to show all columns and full text in dataset
# pd.set_option('display.max_columns', None)
# pd.set_option('display.max_colwidth', None)

In [ ]:
# Data loading in pandas Dataframe
df = pd.read_excel("/content/drive/MyDrive/AI Society Fall23/cleaned_group_seperate.xlsx")
df.head()

web-scraper-order                     web-scraper-start-url  \
0      1722693277-1  https://asu.campuslabs.com/engage/events   
1      1722693286-2  https://asu.campuslabs.com/engage/events   
2      1722693293-3  https://asu.campuslabs.com/engage/events   
3      1722693302-4  https://asu.campuslabs.com/engage/events   
4      1722693309-5  https://asu.campuslabs.com/engage/events   

                   Event_name  \
0  Hudson Park Beautification   
1  Hudson Park Beautification   
2  Hudson Park Beautification   
3  Hudson Park Beautification   
4  Hudson Park Beautification   

                                           date_time  \
0  Date and TimeSaturday, December 6 2025 at 8:00...   
1  Date and TimeSaturday, November 1 2025 at 9:00...   
2  Date and TimeSaturday, October 4 2025 at 9:00 ...   
3  Date and TimeSaturday, September 6 2025 at 9:0...   
4  Date and TimeSaturday, August 2 2025 at 9:00 A...   

                                            location  \
0  LocationHudson Park, 1430 S Cedar St, Tempe, A...   
1  LocationHudson Park, 1430 S Cedar St, Tempe, A...   
2  LocationHudson Park, 1430 S Cedar St, Tempe, A...   
3  LocationHudson Park, 1430 S Cedar St, Tempe, A...   
4  LocationHudson Park, 1430 S Cedar St, Tempe, A...   

                                         description     categories host_org  \
0  DescriptionHudson Park clean up, barbecue fell...  Not Specified      NaN   
1  DescriptionHudson Park clean up, barbecue fell...  Not Specified      NaN   
2  DescriptionHudson Park clean up, barbecue fell...  Not Specified      NaN   
3  DescriptionHudson Park clean up, barbecue fell...  Not Specified      NaN   
4  DescriptionHudson Park clean up, barbecue fell...  Not Specified      NaN   

  event_perks                                         event_link  \
0         NaN  Hudson Park BeautificationSaturday, December 6...   
1         NaN  Hudson Park BeautificationSaturday, November 1...   
2         NaN  Hudson Park BeautificationSaturday, October 4 ...   
3         NaN  Hudson Park BeautificationSaturday, September ...   
4         NaN  Hudson Park BeautificationSaturday, August 2 a...   

                                   event_link-href  \
0  https://asu.campuslabs.com/engage/event/4421709   
1  https://asu.campuslabs.com/engage/event/4421691   
2  https://asu.campuslabs.com/engage/event/4422004   
3  https://asu.campuslabs.com/engage/event/4421987   
4  https://asu.campuslabs.com/engage/event/4421982   

               date_extracted time_extracted  \
0   Saturday, December 6 2025        8:00 AM   
1   Saturday, November 1 2025        9:00 AM   
2    Saturday, October 4 2025        9:00 AM   
3  Saturday, September 6 2025        9:00 AM   
4     Saturday, August 2 2025        9:00 AM   

                              location_extracted  \
0  Hudson Park, 1430 S Cedar St, Tempe, AZ 85281   
1  Hudson Park, 1430 S Cedar St, Tempe, AZ 85281   
2  Hudson Park, 1430 S Cedar St, Tempe, AZ 85281   
3  Hudson Park, 1430 S Cedar St, Tempe, AZ 85281   
4  Hudson Park, 1430 S Cedar St, Tempe, AZ 85281   

                              description_normalized  
0  descriptionhudson park clean up barbecue fello...  
1  descriptionhudson park clean up barbecue fello...  
2  descriptionhudson park clean up barbecue fello...  
3  descriptionhudson park clean up barbecue fello...  
4  descriptionhudson park clean up barbecue fello...

In [ ]:
df.shape

(1888, 15)

In [ ]:
df.describe()

web-scraper-order                     web-scraper-start-url  \
count               1888                                      1888   
unique              1888                                         1   
top         1722693277-1  https://asu.campuslabs.com/engage/events   
freq                   1                                      1888   

                                           Event_name  \
count                                            1888   
unique                                            365   
top     Watts West Wellness Weekly Walking Wanderings   
freq                                               68   

                                                date_time         location  \
count                                                1888              887   
unique                                                803              309   
top     Date and TimeWednesday, August 21 2024 at 6:00...  LocationBAC 316   
freq                                                   13               58   

                                              description     categories  \
count                                                1888           1888   
unique                                                375             32   
top     DescriptionWatts West Wellness Weekly Walking ...  Not Specified   
freq                                                   68            903   

                                    host_org                    event_perks  \
count                                   1778                            803   
unique                                   148                              5   
top     Host Organization TAMID Group at ASU  Perks  Free Food   Free Stuff   
freq                                     155                            427   

                                               event_link  \
count                                                1888   
unique                                                885   
top     Club Hub 2024Thursday, August 22 at 5:00PM MST...   
freq                                                    9   

                                         event_link-href  \
count                                               1888   
unique                                               887   
top     https://asu.campuslabs.com/engage/event/10076957   
freq                                                   9   

                   date_extracted time_extracted location_extracted  \
count                        1888           1888               1888   
unique                        171             34                310   
top     Wednesday, August 21 2024        6:00 PM   Location Unknown   
freq                          129            442               1001   

                                   description_normalized  
count                                                1888  
unique                                                373  
top     descriptionwatts west wellness weekly walking ...  
freq                                                   68

Classes can have attributes and methods, attributes are .word, while methods are .word(), or .word(arg1, arg2, etc.)

You won't know in advance whether something you want to call is a method or attribute, but you will if you read the documentation for that class. In that documentation, shape is listed under attributes not under methods, so you can infer from that classification how to use it (i.e. without parentheses). Here's the doc link for pandas dataframes: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1888 entries, 0 to 1887
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   web-scraper-order       1888 non-null   object
 1   web-scraper-start-url   1888 non-null   object
 2   Event_name              1888 non-null   object
 3   date_time               1888 non-null   object
 4   location                887 non-null    object
 5   description             1888 non-null   object
 6   categories              1888 non-null   object
 7   host_org                1778 non-null   object
 8   event_perks             803 non-null    object
 9   event_link              1888 non-null   object
 10  event_link-href         1888 non-null   object
 11  date_extracted          1888 non-null   object
 12  time_extracted          1888 non-null   object
 13  location_extracted      1888 non-null   object
 14  description_normalized  1888 non-null   object
dtypes: o

In [ ]:
df.isnull().sum()

web-scraper-order            0
web-scraper-start-url        0
Event_name                   0
date_time                    0
location                  1001
description                  0
categories                   0
host_org                   110
event_perks               1085
event_link                   0
event_link-href              0
date_extracted               0
time_extracted               0
location_extracted           0
description_normalized       0
dtype: int64

In [ ]:
df.isnull().sum().sum()

2196


_________
*   We have 15 features in the web scrapped data
*   Let's hand all features one at a time



In [ ]:
# Combine date_extracted and time_extracted to form a datetime column
df['datetime'] = pd.to_datetime(df['date_extracted'] + ' ' + df['time_extracted'], errors='coerce')

# Drop rows where datetime conversion failed
df.dropna(subset=['datetime'], inplace=True)

In [ ]:
df.shape

(1888, 16)

In [ ]:
df.datetime.isnull().sum()

0

In [ ]:
print(df['datetime'].dtype)

datetime64[ns]


* *We created a new feature datetime by combining date and time and converting it into suitable format*
* *Now let's drop unnecessary features.*

In [ ]:
df.head()

web-scraper-order                     web-scraper-start-url  \
0      1722693277-1  https://asu.campuslabs.com/engage/events   
1      1722693286-2  https://asu.campuslabs.com/engage/events   
2      1722693293-3  https://asu.campuslabs.com/engage/events   
3      1722693302-4  https://asu.campuslabs.com/engage/events   
4      1722693309-5  https://asu.campuslabs.com/engage/events   

                   Event_name  \
0  Hudson Park Beautification   
1  Hudson Park Beautification   
2  Hudson Park Beautification   
3  Hudson Park Beautification   
4  Hudson Park Beautification   

                                           date_time  \
0  Date and TimeSaturday, December 6 2025 at 8:00...   
1  Date and TimeSaturday, November 1 2025 at 9:00...   
2  Date and TimeSaturday, October 4 2025 at 9:00 ...   
3  Date and TimeSaturday, September 6 2025 at 9:0...   
4  Date and TimeSaturday, August 2 2025 at 9:00 A...   

                                            location  \
0  LocationHudson Park, 1430 S Cedar St, Tempe, A...   
1  LocationHudson Park, 1430 S Cedar St, Tempe, A...   
2  LocationHudson Park, 1430 S Cedar St, Tempe, A...   
3  LocationHudson Park, 1430 S Cedar St, Tempe, A...   
4  LocationHudson Park, 1430 S Cedar St, Tempe, A...   

                                         description     categories host_org  \
0  DescriptionHudson Park clean up, barbecue fell...  Not Specified      NaN   
1  DescriptionHudson Park clean up, barbecue fell...  Not Specified      NaN   
2  DescriptionHudson Park clean up, barbecue fell...  Not Specified      NaN   
3  DescriptionHudson Park clean up, barbecue fell...  Not Specified      NaN   
4  DescriptionHudson Park clean up, barbecue fell...  Not Specified      NaN   

  event_perks                                         event_link  \
0         NaN  Hudson Park BeautificationSaturday, December 6...   
1         NaN  Hudson Park BeautificationSaturday, November 1...   
2         NaN  Hudson Park BeautificationSaturday, October 4 ...   
3         NaN  Hudson Park BeautificationSaturday, September ...   
4         NaN  Hudson Park BeautificationSaturday, August 2 a...   

                                   event_link-href  \
0  https://asu.campuslabs.com/engage/event/4421709   
1  https://asu.campuslabs.com/engage/event/4421691   
2  https://asu.campuslabs.com/engage/event/4422004   
3  https://asu.campuslabs.com/engage/event/4421987   
4  https://asu.campuslabs.com/engage/event/4421982   

               date_extracted time_extracted  \
0   Saturday, December 6 2025        8:00 AM   
1   Saturday, November 1 2025        9:00 AM   
2    Saturday, October 4 2025        9:00 AM   
3  Saturday, September 6 2025        9:00 AM   
4     Saturday, August 2 2025        9:00 AM   

                              location_extracted  \
0  Hudson Park, 1430 S Cedar St, Tempe, AZ 85281   
1  Hudson Park, 1430 S Cedar St, Tempe, AZ 85281   
2  Hudson Park, 1430 S Cedar St, Tempe, AZ 85281   
3  Hudson Park, 1430 S Cedar St, Tempe, AZ 85281   
4  Hudson Park, 1430 S Cedar St, Tempe, AZ 85281   

                              description_normalized            datetime  
0  descriptionhudson park clean up barbecue fello... 2025-12-06 08:00:00  
1  descriptionhudson park clean up barbecue fello... 2025-11-01 09:00:00  
2  descriptionhudson park clean up barbecue fello... 2025-10-04 09:00:00  
3  descriptionhudson park clean up barbecue fello... 2025-09-06 09:00:00  
4  descriptionhudson park clean up barbecue fello... 2025-08-02 09:00:00

In [ ]:
print(df.iloc[1887,:])

web-scraper-order                                           1722699807-1888
web-scraper-start-url              https://asu.campuslabs.com/engage/events
Event_name                                       Hudson Park Beautification
date_time                 Date and TimeSaturday, August 3 2024 at 9:00 A...
location                  LocationHudson Park, 1430 S Cedar St, Tempe, A...
description               DescriptionHudson Park clean up, barbecue fell...
categories                                                    Not Specified
host_org                                                                NaN
event_perks                                                             NaN
event_link                Hudson Park BeautificationSaturday, August 3 a...
event_link-href             https://asu.campuslabs.com/engage/event/4421763
date_extracted                                      Saturday, August 3 2024
time_extracted                                                      9:00 AM
location_ext

In [ ]:
# dropping irrelevant features.
df.drop(['web-scraper-order', 'web-scraper-start-url', 'event_link', 'event_link-href', 'date_extracted', 'time_extracted', 'date_time', 'description_normalized'], axis=1, inplace=True)

In [ ]:
df.head()

Event_name  \
0  Hudson Park Beautification   
1  Hudson Park Beautification   
2  Hudson Park Beautification   
3  Hudson Park Beautification   
4  Hudson Park Beautification   

                                            location  \
0  LocationHudson Park, 1430 S Cedar St, Tempe, A...   
1  LocationHudson Park, 1430 S Cedar St, Tempe, A...   
2  LocationHudson Park, 1430 S Cedar St, Tempe, A...   
3  LocationHudson Park, 1430 S Cedar St, Tempe, A...   
4  LocationHudson Park, 1430 S Cedar St, Tempe, A...   

                                         description     categories host_org  \
0  DescriptionHudson Park clean up, barbecue fell...  Not Specified      NaN   
1  DescriptionHudson Park clean up, barbecue fell...  Not Specified      NaN   
2  DescriptionHudson Park clean up, barbecue fell...  Not Specified      NaN   
3  DescriptionHudson Park clean up, barbecue fell...  Not Specified      NaN   
4  DescriptionHudson Park clean up, barbecue fell...  Not Specified      NaN   

  event_perks                             location_extracted  \
0         NaN  Hudson Park, 1430 S Cedar St, Tempe, AZ 85281   
1         NaN  Hudson Park, 1430 S Cedar St, Tempe, AZ 85281   
2         NaN  Hudson Park, 1430 S Cedar St, Tempe, AZ 85281   
3         NaN  Hudson Park, 1430 S Cedar St, Tempe, AZ 85281   
4         NaN  Hudson Park, 1430 S Cedar St, Tempe, AZ 85281   

             datetime  
0 2025-12-06 08:00:00  
1 2025-11-01 09:00:00  
2 2025-10-04 09:00:00  
3 2025-09-06 09:00:00  
4 2025-08-02 09:00:00

In [ ]:
df.shape

(1888, 8)

In [ ]:
df.columns

Index(['Event_name', 'location', 'description', 'categories', 'host_org',
       'event_perks', 'location_extracted', 'datetime'],
      dtype='object')

* *We can see that there are two location features.*
* *Let's check other features.*

In [ ]:
df.location.isnull().sum()

1001

In [ ]:
df.location_extracted.isnull().sum()

0

In [ ]:
len(df.location_extracted.unique())

310

In [ ]:
df[df.location.isnull()]['location_extracted']

9       Location Unknown
11      Location Unknown
13      Location Unknown
16      Location Unknown
18      Location Unknown
              ...       
1880    Location Unknown
1882    Location Unknown
1884    Location Unknown
1885    Location Unknown
1886    Location Unknown
Name: location_extracted, Length: 1001, dtype: object

In [ ]:
df[df.location.isnull()]['location_extracted'].unique()

array(['Location Unknown'], dtype=object)

In [ ]:
df.location_extracted = df.location_extracted.str.replace("Location Unknown", "Unknown")

In [ ]:
df.drop('location', axis=1, inplace=True)

In [ ]:
# Fill missing values in 'host_org' with 'Unknown'
df['host_org'].fillna('Unknown', inplace=True)

# Fill missing values in 'event_perks' with 'None'
df['event_perks'].fillna('None', inplace=True)

In [ ]:
df.isnull().sum()

Event_name            0
description           0
categories            0
host_org              0
event_perks           0
location_extracted    0
datetime              0
dtype: int64

In [ ]:
df[['categories','host_org','event_perks','location_extracted']].describe()

categories                              host_org event_perks  \
count            1888                                  1888        1888   
unique             32                                   149           6   
top     Not Specified  Host Organization TAMID Group at ASU        None   
freq              903                                   155        1085   

       location_extracted  
count                1888  
unique                310  
top               Unknown  
freq                 1001

In [ ]:
df.head()

Event_name  \
0  Hudson Park Beautification   
1  Hudson Park Beautification   
2  Hudson Park Beautification   
3  Hudson Park Beautification   
4  Hudson Park Beautification   

                                         description     categories host_org  \
0  DescriptionHudson Park clean up, barbecue fell...  Not Specified  Unknown   
1  DescriptionHudson Park clean up, barbecue fell...  Not Specified  Unknown   
2  DescriptionHudson Park clean up, barbecue fell...  Not Specified  Unknown   
3  DescriptionHudson Park clean up, barbecue fell...  Not Specified  Unknown   
4  DescriptionHudson Park clean up, barbecue fell...  Not Specified  Unknown   

  event_perks                             location_extracted  \
0        None  Hudson Park, 1430 S Cedar St, Tempe, AZ 85281   
1        None  Hudson Park, 1430 S Cedar St, Tempe, AZ 85281   
2        None  Hudson Park, 1430 S Cedar St, Tempe, AZ 85281   
3        None  Hudson Park, 1430 S Cedar St, Tempe, AZ 85281   
4        None  Hudson Park, 1430 S Cedar St, Tempe, AZ 85281   

             datetime  
0 2025-12-06 08:00:00  
1 2025-11-01 09:00:00  
2 2025-10-04 09:00:00  
3 2025-09-06 09:00:00  
4 2025-08-02 09:00:00

In [ ]:
df.shape

(1888, 7)

* **I suspect dublicate values, let's investigate the hypothesis.**

In [ ]:
df.Event_name.value_counts()

Event_name
Watts West Wellness Weekly Walking Wanderings    68
TAMID Group General Meetings                     66
Thursday Discussion Forum                        65
Tuesday Discussion Forum                         60
TAMID Group Fund Meetings                        58
                                                 ..
Corn Maze                                         1
Second degree celebration                         1
Sports Business Scholars First Meeting 2024       1
Panhellenic Spin Class                            1
Meet the Zetas: Cupcake Decorating                1
Name: count, Length: 365, dtype: int64

In [ ]:
df.Event_name.value_counts()[df.Event_name.value_counts() > 2]

Event_name
Watts West Wellness Weekly Walking Wanderings                       68
TAMID Group General Meetings                                        66
Thursday Discussion Forum                                           65
Tuesday Discussion Forum                                            60
TAMID Group Fund Meetings                                           58
                                                                    ..
HOSA St. Mary's Volunteer Event                                      3
Sun Devil Gaming LAN                                                 3
Glo BINGO Hosted by the College of Integrative Sciences and Arts     3
Welcome Back                                                         3
Fall Carnival                                                        3
Name: count, Length: 191, dtype: int64

In [ ]:
df.groupby('Event_name')['datetime'].value_counts()[df.groupby('Event_name')['datetime'].value_counts() > 1]

Event_name                                                          datetime           
11th Annual Healthcare Panel Banquet                                2024-11-14 18:00:00    3
11th Annual Poly Game Night                                         2024-08-23 18:00:00    4
2024 Annual SunMUN High School Conference                           2024-11-15 08:00:00    4
A.T. Still University Campus Visit- Pre health programs             2024-10-18 12:00:00    2
ACF weekly event                                                    2024-08-03 12:00:00    4
                                                                                          ..
Welcome to West Valley!                                             2024-08-17 18:00:00    4
West Fest                                                           2024-08-21 17:00:00    3
West Valley goes BIG (12)                                           2024-08-18 19:00:00    4
What is SoDA and Can I Drink It?                                    2024-08-27 19:30:00    8
What is TRIO Teacher Prep (and why should I attend this workshop)?  2024-08-07 16:00:00    2
Name: count, Length: 447, dtype: int64

In [ ]:
df[df.Event_name == "11th Annual Healthcare Panel Banquet"]

Event_name  \
279  11th Annual Healthcare Panel Banquet   
280  11th Annual Healthcare Panel Banquet   
281  11th Annual Healthcare Panel Banquet   

                                           description       categories  \
279  DescriptionThe Black Medical Students Associat...    Not Specified   
280  DescriptionThe Black Medical Students Associat...    Club Meetings   
281  DescriptionThe Black Medical Students Associat...  In-Person Event   

                                              host_org  \
279  Host Organization Black Medical Student Associ...   
280  Host Organization Black Medical Student Associ...   
281  Host Organization Black Medical Student Associ...   

                       event_perks     location_extracted            datetime  
279  Perks  Free Food   Free Stuff  Memorial Union MU-202 2024-11-14 18:00:00  
280  Perks  Free Food   Free Stuff                Unknown 2024-11-14 18:00:00  
281  Perks  Free Food   Free Stuff                Unknown 2024-11-14 18:00:00

Same event is repeated thrice, With different categories and the location is mentioned only once.

In [ ]:
df.shape

(1888, 7)

In [ ]:
# Group by 'Event_name' and 'datetime', then aggregate categories and location_extracted
result = df.groupby(['Event_name', 'datetime', 'description']).agg({
    'host_org': lambda x: ', '.join(x[x != 'Unknown'].unique()),
    'event_perks': lambda x: ', '.join(x[x != 'None'].unique()),
    'categories': lambda x: ', '.join(x[x != "Not Specified"].unique()),
    'location_extracted': lambda x: ', '.join(x[x != 'Unknown'].unique())  # Exclude 'Unknown' locations
}).reset_index()

In [ ]:
result.shape

(878, 7)

In [ ]:
# Rename columns
result.columns = ['event_name', 'datetime', 'description', 'host_org', 'event_perks', 'categories', 'location']
result.head()

event_name            datetime  \
0  'These Are Your First Steps': A Brief Introduc... 2024-09-18 10:00:00   
1               11th Annual Healthcare Panel Banquet 2024-11-14 18:00:00   
2                        11th Annual Poly Game Night 2024-08-23 18:00:00   
3          2024 Annual SunMUN High School Conference 2024-11-15 08:00:00   
4             7V7 and Powderpuff Football Tournament 2024-08-29 17:00:00   

                                         description  \
0  DescriptionAre you interested in learning how ...   
1  DescriptionThe Black Medical Students Associat...   
2  DescriptionJoin ASU Library for the 11th Annua...   
3  DescriptionDelegates from across Arizona and t...   
4  DescriptionA football tournament where men and...   

                                            host_org  \
0                      Host Organization ASU Library   
1  Host Organization Black Medical Student Associ...   
2    Host Organizations ASU Library  ASU Polytechnic   
3  Host Organization Model United Nations at Ariz...   
4      Host Organization HOPE CHURCH MOVEMENT AT ASU   

                     event_perks  \
0                                  
1  Perks  Free Food   Free Stuff   
2  Perks  Free Food   Free Stuff   
3                                  
4                                  

                                      categories  \
0                                                  
1                 Club Meetings, In-Person Event   
2          General, PAB Event, ASU Welcome Event   
3  Club Meetings, International, In-Person Event   
4                                                  

                                            location  
0                                             Online  
1                              Memorial Union MU-202  
2  Polytechnic campus Library5988 S. Backus Mall,...  
3  Arizona State University Tempe Campus1151 S Fo...  
4                                    SDFC West Field

In [ ]:
result.isnull().sum()

event_name     0
datetime       0
description    0
host_org       0
event_perks    0
categories     0
location       0
dtype: int64

In [ ]:
result.event_perks.value_counts()

event_perks
                                 657
Perks  Free Food   Free Stuff    104
Perks  Free Food                  72
Perks  Free Stuff                 43
Perks  Credit   Free Food          1
Perks  Credit                      1
Name: count, dtype: int64

In [ ]:
result.replace('', np.nan, inplace=True)
result.isnull().sum()

event_name       0
datetime         0
description      0
host_org        63
event_perks    657
categories     436
location         0
dtype: int64

In [ ]:
result.groupby('event_name')['datetime'].value_counts()[result.groupby('event_name')['datetime'].value_counts() > 1]

event_name                    datetime           
ASU United Worship Gathering  2024-08-22 19:00:00    2
Major Monday                  2024-08-26 10:00:00    2
Name: count, dtype: int64

In [ ]:
result[result.event_name == "ASU United Worship Gathering"]

event_name            datetime  \
28  ASU United Worship Gathering 2024-08-22 19:00:00   
29  ASU United Worship Gathering 2024-08-22 19:00:00   

                                          description  \
28  DescriptionCome together with all the other ca...   
29  DescriptionCome together with all the other ca...   

                                             host_org event_perks  \
28  Host Organization InterVarsity Christian Fello...         NaN   
29  Host Organization InterVarsity Christian Fello...         NaN   

           categories                                           location  
28  ASU Welcome Event  Neeb Hall L. S. Neeb Hall, 920 S Forest Mall, ...  
29  ASU Welcome Event  Neeb Hall Neeb Hall, 920 S Forest Mall, , Temp...

In [ ]:
result[result.event_name == "Major Monday"]

event_name            datetime  \
392  Major Monday 2024-08-26 10:00:00   
393  Major Monday 2024-08-26 10:00:00   

                                           description  \
392  Description\nIs your major a good fit?\n\nCome...   
393  DescriptionIs your major the right fit?\n- Lea...   

                                              host_org  \
392  Host Organizations New College of Interdiscipl...   
393  Host Organizations New College of Interdiscipl...   

                       event_perks  \
392              Perks  Free Stuff   
393  Perks  Free Food   Free Stuff   

                                            categories  \
392  Career and Professional Development, General, ...   
393                                  ASU Welcome Event   

                                              location  
392  Fletcher Lawn4701 W. Thunderbird Road, Glendal...  
393  Fletcher Lawn4701 W Thunderbird Rd, Glendale, ...

In [ ]:
result.shape

(878, 7)

In [ ]:
result = result.drop([392, 29])
result.reset_index(drop=True, inplace=True)

In [ ]:
result.shape

(876, 7)

In [ ]:
result.location.value_counts()

location
BAC 316                                                                                  58
Online                                                                                   33
BA 259                                                                                   29
Business Administration (BA) 199                                                         27
TBD                                                                                      26
                                                                                         ..
Polytechnic, Cooley BC                                                                    1
Walton Center for Planetary Health - Tempe Campus777 E. University Dr., TempeView Map     1
Golf Land, Adress: 155 W Hampton Ave, Mesa, AZ 85210                                      1
MU Cochise 228301 E Orange St., Tempe, AZ 85281View Map                                   1
Lost Canyon                                                                               1
Name: count, Length: 300, dtype: int64

In [ ]:
result['location'] = result['location'].str.replace('View Map', '')
result.location.unique()

array(['Online', 'Memorial Union MU-202',
       'Polytechnic campus Library5988 S. Backus Mall,  Mesa, AZ',
       'Arizona State University Tempe Campus1151 S Forest Avenue, Tempe, AZ',
       'SDFC West Field',
       'AT Still University5850 E. Still Circle, Mesa, AZ',
       'Christian Challenge1322 S Mill Ave, Tempe, Arizona', 'LSA 101',
       'Stauffer B125950 S. Forest Mall Tempe, AZ 85281, Tempe, Arizona',
       'GLCC 206/208 AB',
       'All Saints Catholic Newman Center230 E. University Dr, Tempe, Arizona',
       'Chase Field - Phoenix',
       'Design North Room CND60810 S Forest Mall  Room 60, Tempe, AZ',
       'Memorial Union301 E Orange St., Tempe, AZ 85281',
       'Neeb Hall L. S. Neeb Hall, 920 S Forest Mall, , Tempe, Arizona',
       'Papago Park, Adress: 625 N Galvin Pkwy, Phoenix, AZ 85008',
       'ASU Memorial Union301 East Orange Street, Tempe, Arizona',
       'Memorial Union #226 (Graham Room)301 E Orange St., Tempe, AZ 85281',
       'BAC 215400 E Lemon S

_______
* We have removed dublicates successfully
* We have no missing values in the location feature

In [ ]:
result.event_name.value_counts()

event_name
TAMID Group General Meetings                                        33
TAMID Group Fund Meetings                                           29
TAMID Group Executive Board Meetings                                25
Chapter                                                             24
Business School Council General Meeting                             18
                                                                    ..
Gold Friday 8/23                                                     1
Global Futures Opportunities Fair                                    1
Glo BINGO Hosted by the College of Integrative Sciences and Arts     1
Ganesh Vinayagar Chaturthi                                           1
Young Life College Weekend                                           1
Name: count, Length: 365, dtype: int64

__________________________
* We have 365 total unique events and a total of 876 events.

In [ ]:
result.columns

Index(['event_name', 'datetime', 'description', 'host_org', 'event_perks',
       'categories', 'location'],
      dtype='object')

In [ ]:
# Let's export the data for later use.
result.to_csv('cleaned_data.csv', index=False)

* We will continue in next session.